node/libs/roles/src/validator/messages/consensus.rs

```diff
@@ -74,25 +76,48 @@ impl Default for Fork {
 /// A struct that represents a set of validators. It is used to store the current validator set.
 /// We represent each validator by its validator public key.
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub struct ValidatorSet {
+pub struct ValidatorCommittee {
     vec: Vec<validator::PublicKey>,
-    map: BTreeMap<validator::PublicKey, usize>,
+    indexes: BTreeMap<validator::PublicKey, usize>,
+    weights: Vec<usize>,
 }
 
-impl ValidatorSet {
-    /// Creates a new ValidatorSet from a list of validator public keys.
-    pub fn new(validators: impl IntoIterator<Item = validator::PublicKey>) -> anyhow::Result<Self> {
+impl ValidatorCommittee {
+    /// Maximum validator weight
+    /// 100.00%
+    pub const MAX_WEIGHT: usize = 10000;
+
```

Comment:
why not uint

Reply:
also all the arithmetic on this type should be checked for overflows

Reply:
switched all weights to u64, and also removed this constant as it is no longer necessary